pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/GenericSigBase.java

```diff
@@ -250,5 +255,16 @@ public JTypeMirror getReturnType() {
         public String toString() {
             return signature;
         }
+
+        void addParameterAnnotation(int paramIndex, SymAnnot annot) {
+            if (parameterAnnotations.isEmpty()) {
```

Comment:
One thing is the annotations you are adding here are the annotations on the variable, not the type annotations. Accordingly, they shouldn't be in the method signature, but stay in the symbol (ExecutableStub)

Reply:
See 2df039f4872d1fd6c925be2e5e08da76ddee87c8 for the relevant changes

Reply:
Great point, this should be fixed now.